### DISCUSSION TITLE
How to painlessly move autodiscovered device from z2m(1) to z2m(2) without HA to know

### DISCUSSION THREAD

**Discussion by mr2c12** [Koenkk/zigbee2mqtt#10863] [Devices]
My current setup involves a z2m (name it "A") network with a lot of devices, over two different floors (coordinator is downstairs with lots of router/end devices, while upstairs i have other routers/end devices).
Quite often I experience some troubles with upstairs devices, so I am now convinced to create a secondary z2m (name it "B") on that floor.

I'm aware I will need to re-pair everything and use a different channel, that's fine.
The problem comes with Home Assistant.
Most of those devices are manually configured, so it's quite easy to trick HA: I just rename the mqtt topic in my yaml files, that's all.
When it comes to autodiscovered devices, I'm quite lost. I cannot find any reference to the current mqtt topics, so I suppose I need to delete the existing HA entities and re-add them through z2m (B).
Since my scripts and automations are contained in yaml files, I'm going to loose for sure:
- customization made with the UI
- other?

With autodiscovered devices, how can I move devices from z2m(A) to z2m(B) and get the less troubles as possible in HA?

**Comment by Koenkk** [comment]
I haven't tested it myself, but I think the following will work
- Stop HA
- Force remove the device from z2m A
- Pair the device to z2m B
- Start HA

**Reply by mr2c12** [reply]
My concern is how the "unique id" is built on z2m.
I'm actually away from my pc, but it should contain the device id and the "zigbee2mqtt" part. Is "zigbee2mqtt" fixed or it's the base topic for the current z2m instance?

**Reply by Koenkk** [reply]
@mr2c12 only for groups the discovery topic contains the z2m base topic, for devices it is it the ieeeAddress: https://github.com/Koenkk/zigbee2mqtt/blob/1a0a9a6eabf58855d3a6feda60bb79a1c34d17f8/lib/extension/homeassistant.ts#L1330
